solutions/USACO dec15-high-card-low-card-gold.mdx

```diff
@@ -0,0 +1,108 @@
+---
+id: usaco-573
+source: USACO Gold 2015 December
+title: High Card Low Card (Gold)
+author: Qi Wang
+---
+
+**Time Complexity:** $\mathcal{O}(N\logN)$
+
+[Official Analysis](http://www.usaco.org/current/data/sol_cardgame_gold_dec15.html)
+
+<LanguageSection>
+
+<JavaSection>
+
+```java
+
+import java.util.*;
+import java.io.*;
+
+public class highcardgold {
+	static int[] C;
+	static List<Integer> B = new ArrayList<>();
+	static List<Integer> F = new ArrayList<>();
+	static List<Integer> S = new ArrayList<>();
+	static int N;
+	public static void main(String[] args) throws IOException {
+		Kattio io = new Kattio("cardgame");
+		N = io.nextInt();
+
+		C = new int[2*N+1];
+
+		for (int i = 0; i < N; i++) {
+			int c = io.nextInt();
+			C[c]++;
+			if(i < N/2)
+				F.add(c);
+			else
+				S.add(c);
+		}
+
+		for (int i = 1; i < 2 * N + 1; i++) {
+			if(C[i] == 0) B.add(i);
+		}
+
+		Collections.sort(F);
+		Collections.sort(B);
+		Collections.sort(S);
+
+		int ix = B.size()-1;
+		int score = 0;
+		for (int i = F.size()-1; i >= 0; i--) {
+			if(ix >= 0 && F.get(i) < B.get(ix)){
+				score++;
+				B.remove(ix);
+				ix--;
+			}
+		}
+
+		ix = 0;
+		for (int i = 0; i < S.size(); i++) {
+			if(ix < B.size() && B.get(ix) < S.get(i)){
+				ix++;
+				score++;
+			}
+		}
```

Comment:
I'm fairly certain `ix >= 0`, `B.remove(ix);`, and `ix < B.size()` are unnecessary. Whether or not you should remove them is up to you.

Reply:
true did not notice!!